packages/front-end/components/Settings/EditDataSourceSettingsForm.tsx

```diff
@@ -444,13 +451,15 @@ const EditDataSourceSettingsForm: FC<{
               <div className="bg-light border my-2 p-3 ml-3">
                 <div className="row mb-3">
                   <div className="col">
-                    <Field
+                    <CodeTextArea
                       label="Python runQuery definition"
+                      language="python"
                       placeholder="def runQuery(sql):"
                       {...form.register("settings.notebookRunQuery")}
-                      textarea
-                      minRows={5}
-                      maxRows={20}
+                      value={form.watch(`settings.notebookRunQuery`)}
+                      setValue={(sql) =>
+                        form.setValue(`settings.notebookRunQuery`, sql)
```

Comment:
This variable name should not be `sql` since it's dealing with python code instead

Reply:
DOH! Embarrassing. I've updated this. I also realized that line 458 (I'm pretty sure) is duplicate and not necessary. So I've removed.